xds/src/main/java/io/grpc/xds/XdsClusterResource.java

```diff
@@ -282,16 +241,41 @@ private static StructOrError<CdsUpdate.Builder> parseNonAggregateCluster(
           + ": transport-socket-matches not supported.");
     }
     if (cluster.hasTransportSocket()) {
-      if (!TRANSPORT_SOCKET_NAME_TLS.equals(cluster.getTransportSocket().getName())) {
-        return StructOrError.fromError("transport-socket with name "
-            + cluster.getTransportSocket().getName() + " not supported.");
+      String transportSocketName = cluster.getTransportSocket().getName();
+
+      if (!TRANSPORT_SOCKET_NAME_TLS.equals(transportSocketName) &&
+          (isEnabledXdsHttpConnect && !TRANSPORT_SOCKET_NAME_HTTP11_PROXY.equals(transportSocketName))) {
+        return StructOrError.fromError(
+            "Transport socket with name " + transportSocketName + " not supported.");
+      }
+
+      Any tlsConfig;
+      if (isEnabledXdsHttpConnect && TRANSPORT_SOCKET_NAME_HTTP11_PROXY.equals(transportSocketName)) {
+        isHttp11ProxyEnabled = true;
+        try {
+          tlsConfig = unpackCompatibleType(
```

Comment:
unpackCompatibleType() is when there are multiple messages that are compatible. Here, you only have a single message type, so use any.unpack().